### PR TITLE
upgrade castle integration

### PR DESCRIPTION
**Pull request by bartes**
**What does this PR do?**
- switched castle integration to the new castle script (https://www.npmjs.com/package/@castleio/castle-js).  old one (https://www.npmjs.com/package/castle.js) is no longer maintained. Maps segment.page to https://docs.castle.io/docs/sdk-browser#castlepage and segment.track to https://docs.castle.io/docs/sdk-browser#castlecustom .

**Are there breaking changes in this PR?**
 - no , legacy integrations should work without any changes

**Testing**
Testing completed successfully using unit test, and by updating 2 integrations with the dev:build
-  test suite has been updated and extended accordingly

**Any background context you want to provide?**
- this PR switches setup to the new script which includes additional features

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
- not yet 

**Does this require a new integration setting? If so, please explain how the new setting works**
- no (but some legacy options are no longer needed, and were removed)

**Links to helpful docs and other external resources**
- https://docs.castle.io/docs/sdk-browser

### CODE REVIEW

File: integrations/castle/lib/index.js [hunk 1]
```diff
@@ -5,39 +5,44 @@
...
 /**
  * Expose `Castle` integration.
  */
 
-var Castle = (module.exports = integration('Castle')
+var CastleIntegration = (module.exports = integration('Castle')
+  .global('CastleSegment')
+  .global('casData')
   .option('publishableKey', '')
-  .option('autoPageview', false)
```
Comment (chrisradek) [inline integrations/castle/lib/index.js:15]: This functionality is no longer supported in the new version?

File: integrations/castle/lib/index.js [hunk 2]
```diff
@@ -5,39 +5,44 @@
...
-  .option('cookieDomain', false)
-  .tag('<script src="//d2t77mnxyo7adj.cloudfront.net/v1/cs.js">'));
+  .option('cookieDomain', '')
```
Comment (chrisradek) [inline integrations/castle/lib/index.js:22]: Awesome, looks like this already shows up as a string in the destination settings page so this change is great!

**Review by chrisradek — COMMENTED** [review]
@bartes Thanks for opening this PR and sorry for the delay in getting someone to review it.

Overall this looks great. The HISTORY.md needs a minor update and the PR needs a merge from the master branch due to some conflicts but otherwise I don't see any issues with these changes!

File: integrations/castle/lib/index.js (hunk 1 above)
Comment (bartes) [reply integrations/castle/lib/index.js:15]: yes it's not supported anymore. I'm not sure, Should make a dummy placeholder option just to cover this ?

Comment (bartes) [reply integrations/castle/lib/index.js:15]: let's keep it removed :)

**Comment by bartes** [comment]
> @bartes Thanks for opening this PR and sorry for the delay in getting someone to review it.
> 
> Overall this looks great. The HISTORY.md needs a minor update and the PR needs a merge from the master branch due to some conflicts but otherwise I don't see any issues with these changes!

Great, thanks for the review. The requested updates have been applied.

**Review by chrisradek — APPROVED** [review]
This looks great! I think we need 1 more merge from the master branch but then we can merge this in.

**Comment by bartes** [comment]
thanks @chrisradek - and it's done 👍

**Comment by bartes** [comment]
Hi @chrisradek . Not sure what is the merging process looks like. Do you think we could have this merged this week ?

**Comment by chrisradek** [comment]
Merging into a feature branch to trigger ci, then will merge into master!

**Comment by chrisradek** [comment]
@bartes This has now been merged into master via #700 

We're currently in the middle of deploying updates so this change will be deployed either later this week or early next week.

**Pull request merged**